Cannot build oils-for-unix-0.26.0 on macOS: time: illegal option -- f

macOS's /usr/bin/time does not support -f option.

<details>
<summary>Full logs:</summary>

```
~/Downloads/oils-for-unix-0.26.0 took 3s
❯ ./_build/oils.sh

./_build/oils.sh: Building oils-for-unix: _bin/cxx-opt-sh/oils-for-unix
    PWD = /Users/lxy/Downloads/oils-for-unix-0.26.0
    cxx = cxx
    variant = opt
    translator = mycpp

CXX _gen/bin/oils_for_unix.mycpp.cc
CXX _gen/bin/text_files.cc
time: illegal option -- f
usage: time [-al] [-h | -p] [-o file] utility [argument ...]
CXX _gen/core/runtime.asdl.cc
CXX _gen/core/value.asdl.cc
CXX _gen/cpp/build_stamp.cc
CXX _gen/data_lang/nil8.asdl.cc
CXX _gen/display/pretty.asdl.cc
CXX _gen/frontend/arg_types.cc
CXX _gen/frontend/consts.cc
CXX _gen/frontend/help_meta.cc
CXX _gen/frontend/id_kind.asdl.cc
CXX _gen/frontend/signal.cc
CXX _gen/frontend/syntax.asdl.cc
CXX _gen/osh/arith_parse.cc
CXX _gen/ysh/grammar_tables.cc
CXX cpp/core.cc
CXX cpp/data_lang.cc
In file included from cpp/data_lang.cc:5:
/Users/lxy/Downloads/oils-for-unix-0.26.0/data_lang/j8.h:76:15: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
   76 |       int n = sprintf((char*)*p_out, "\\u{%x}", ch);
      |               ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:274:1: note: 'sprintf' has been explicitly marked deprecated here
  274 | __deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
      | ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:218:48: note: expanded from macro '__deprecated_msg'
  218 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
      |                                                       ^
In file included from cpp/data_lang.cc:5:
/Users/lxy/Downloads/oils-for-unix-0.26.0/data_lang/j8.h:81:15: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
   81 |       int n = sprintf((char*)*p_out, "\\u%04x", ch);
      |               ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:274:1: note: 'sprintf' has been explicitly marked deprecated here
  274 | __deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
      | ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:218:48: note: expanded from macro '__deprecated_msg'
  218 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
      |                                                       ^
In file included from cpp/data_lang.cc:5:
/Users/lxy/Downloads/oils-for-unix-0.26.0/data_lang/j8.h:108:13: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  108 |     int n = sprintf((char*)*p_out, "\\y%02x", ch);
      |             ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:274:1: note: 'sprintf' has been explicitly marked deprecated here
  274 | __deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
      | ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:218:48: note: expanded from macro '__deprecated_msg'
  218 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
      |                                                       ^
In file included from cpp/data_lang.cc:5:
/Users/lxy/Downloads/oils-for-unix-0.26.0/data_lang/j8.h:112:13: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  112 |     int n = sprintf((char*)*p_out, "\\u%04x", result.codepoint);
      |             ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:274:1: note: 'sprintf' has been explicitly marked deprecated here
  274 | __deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
      | ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:218:48: note: expanded from macro '__deprecated_msg'
  218 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
      |                                                       ^
In file included from cpp/data_lang.cc:5:
/Users/lxy/Downloads/oils-for-unix-0.26.0/data_lang/j8.h:162:13: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  162 |     int n = sprintf((char*)*p_out, "\\u%04x", ch);
      |             ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:274:1: note: 'sprintf' has been explicitly marked deprecated here
  274 | __deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
      | ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:218:48: note: expanded from macro '__deprecated_msg'
  218 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
      |                                                       ^
In file included from cpp/data_lang.cc:5:
/Users/lxy/Downloads/oils-for-unix-0.26.0/data_lang/j8.h:180:13: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  180 |     int n = sprintf((char*)*p_out, "\\x%02x", **p_in);
      |             ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:274:1: note: 'sprintf' has been explicitly marked deprecated here
  274 | __deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
      | ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:218:48: note: expanded from macro '__deprecated_msg'
  218 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
      |                                                       ^
6 warnings generated.
CXX cpp/fanos.cc
CXX cpp/fanos_shared.c
clang++: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
CXX cpp/frontend_flag_spec.cc
CXX cpp/frontend_match.cc
CXX cpp/frontend_pyreadline.cc
cpp/frontend_pyreadline.cc:21:18: warning: unused variable 'gReadline' [-Wunused-variable]
   21 | static Readline* gReadline = nullptr;
      |                  ^~~~~~~~~
cpp/frontend_pyreadline.cc:94:13: warning: unused function 'readline_cb' [-Wunused-function]
   94 | static void readline_cb(char* line) {
      |             ^~~~~~~~~~~
2 warnings generated.
CXX cpp/libc.cc
CXX cpp/osh.cc
CXX cpp/osh_tdop.cc
CXX cpp/pgen2.cc
CXX cpp/pylib.cc
CXX cpp/stdlib.cc
CXX mycpp/bump_leak_heap.cc
CXX mycpp/gc_builtins.cc
mycpp/gc_builtins.cc:117:7: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  117 |       sprintf(p, "\\x%02x", c & 0xff);
      |       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:274:1: note: 'sprintf' has been explicitly marked deprecated here
  274 | __deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
      | ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:218:48: note: expanded from macro '__deprecated_msg'
  218 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
      |                                                       ^
mycpp/gc_builtins.cc:185:53: warning: 'static_assert' with no message is a C++17 extension [-Wc++17-extensions]
  185 |   static_assert(sizeof(long long) == sizeof(int64_t));
      |                                                     ^
      |                                                     , ""
2 warnings generated.
CXX mycpp/gc_iolib.cc
CXX mycpp/gc_mops.cc
CXX mycpp/gc_mylib.cc
CXX mycpp/gc_str.cc
CXX mycpp/hash.cc
CXX mycpp/mark_sweep_heap.cc
WAIT
LINK _bin/cxx-opt-sh/oils-for-unix
clang++: error: no such file or directory: '_build/obj/cxx-opt-sh/_gen/bin/oils_for_unix.o'
```

</details>

--- a/build/ninja-rules-cpp.sh
+++ b/build/ninja-rules-cpp.sh
@@ -330,11 +330,21 @@ compile_one() {
         # Must have external 'time', and it must have -f
         # Notes: OS X has no time -f
         #        busybox time supports -f but not --format.
-        if command time -f '%e %M' true >/dev/null; then
-          set -- \
-            time -f "$out { elapsed: %e, max_RSS: %M }" -- \
-            "$@"
-        fi
+        case ${BASH_VERSION:-} in
+          3*)
+            # bash 3.2.x on macos has a bizarre bug that leads to weird
+            # interactions between errexit and `command time -f` when this
+            # function gets run as a background job (like it does in the
+            # tarball build script). Don't bother timing in this case.
+            ;;
+          *)
+            if command time -f '%e %M' true >/dev/null; then
+              set -- \
+                time -f "$out { elapsed: %e, max_RSS: %M }" -- \
+                "$@"
+            fi
+            ;;
+          esac
         ;;
     esac
   fi

